src/Symfony/Component/Messenger/Worker.php

```diff
@@ -48,7 +65,81 @@ public function run()
                 return;
             }
 
-            $this->bus->dispatch($envelope->with(new ReceivedStamp()));
+            $this->dispatchEvent(
+                WorkerMessageHandlingEvent::class,
+                new WorkerMessageHandlingEvent($envelope, $this->receiverName)
+            );
+
+            $message = $envelope->getMessage();
+            $context = [
+                'message' => $message,
+                'class' => \get_class($message),
+            ];
+
+            try {
```

Comment:
There is a try-catch around the bus, which means if an exception is thrown there, we're handled.

However, there is currently no try-catch around this *whole* thing - around the *whole* `$this->receiver->receive` call. This means that any exceptions thrown by the receiver while reading/decoding the message (specifically `TransportException` or a decoding exception) are not caught. 

Should we catch those and, for example, dispatch another exception? Or should we let that exception be throw, which would naturally trigger the console exception event, and cause the `messenger:consume` command to exit? Exiting is not the end-of-the-world, as the user will need something keep workers running anyways (e.g. supervisord)

Reply:
I'd let it throw, I'm not sure what would we do otherwise anyway?